### PR TITLE
Stdio lib

### PR DESCRIPTION
**Pull request by Plunkerusr**

### CODE REVIEW

File: libs/libsystem/malloc.h [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#define MAX_ALLOCATED_BLOCKS 64
+
+typedef struct malloc_header {
+    size_t size;
+    bool free;
+    struct malloc_header* next;
+    struct malloc_header* prev;
+} malloc_header_t;
+
+void* memory[MAX_ALLOCATED_BLOCKS]; // keeps pointers to the memmory, allocated by mmap
```
Comment (nimelehin) [inline libs/libsystem/malloc.h:17]: replace them with `extern void* memory[MAX_ALLOCATED_BLOCKS];` if other parts should have access to them. Or just put them in .c file

File: libs/libsystem/malloc.c [hunk 2]
```diff
@@ -0,0 +1,99 @@
+#include "malloc.h"
+#include "syscalls.h"
+#include "string.h"
+
```
Comment (nimelehin) [inline libs/libsystem/malloc.c:4]: void* memory[MAX_ALLOCATED_BLOCKS];

File: libs/libsystem/malloc.h [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ uint8_t allocated_blocks = 0;
+
+void* malloc(size_t);
+void free(void*);
+void* kalloc(size_t, size_t);
+void* realloc(void*, size_t);
+
+void __alloc_new_block(size_t);
```
Comment (nimelehin) [inline libs/libsystem/malloc.h:25]: since it's private, leave the definition in .c file

Comment (Plunkerusr) [reply libs/libsystem/malloc.h:25]: ok)

File: libs/libsystem/stdio.c [hunk 4]
```diff
@@ -1,10 +1,29 @@
 #include "stdio.h"
 #include "syscalls.h"
 
-FILE *fopen(const char *filename, const char *mode) {
-    int fd = open(filename, 1);
-    FILE opened_file;
-    opened_file.fd = fd;
-    max_opened_files[opened_file_pointer] = opened_file;
-    return &max_opened_files[opened_file_pointer++];
+size_t _findFreeSlot() {
```
Comment (nimelehin) [inline libs/libsystem/stdio.c:4]: static size_t _find_free_slot

File: src/kernel/interrupts/handlers/sys_handler.c [hunk 5]
```diff
@@ -189,10 +193,50 @@ void sys_mmap(trapframe_t* tf)
...
+void sys_lseek(trapframe_t* tf) 
+{
+    file_descriptor_t* fd = (file_descriptor_t*)proc_get_fd(tasking_get_active_proc(), (int)param1);
+    if (!fd) {
+        // fd is not an open file descriptor.
+        set_return(tf, -EBADF);
+        return;
+    }
+
+    if ((uint32_t)param3 > 2 && (uint32_t)param3 < 0) {
```
Comment (nimelehin) [inline src/kernel/interrupts/handlers/sys_handler.c:214]: Not need to check uint32_t < 0

File: libs/libsystem/malloc.c [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
```
Comment (nimelehin) [inline libs/libsystem/malloc.c:4]: Add definition of `__alloc_new_block` here

File: libs/libsystem/malloc.h [hunk 7]
```diff
@@ -0,0 +1,23 @@
+#ifndef __oneOS__libsystem__MALLOC_H
```
Comment (nimelehin) [inline libs/libsystem/malloc.h:1]: `__oneOS__LibC__MALLOC_H `

File: libs/libsystem/string.h [hunk 8]
```diff
@@ -0,0 +1,11 @@
+#ifndef __oneOS__libsystem__STRING_H
```
Comment (nimelehin) [inline libs/libsystem/string.h:1]: `__oneOS__LibC__STRING_H `

**Review by nimelehin — CHANGES_REQUESTED** [review]

File: libs/libsystem/malloc.c (hunk 6 above)
Comment (Plunkerusr) [reply libs/libsystem/malloc.c:4]: on tyt i est'

**Review by nimelehin — APPROVED** [review]

**Pull request merged**